libraries/shared/watcher/storage_watcher_test.go

```diff
@@ -43,170 +38,223 @@ var _ = Describe("Storage Watcher", func() {
-		w := watcher.NewStorageWatcher(mockTailer, test_config.NewTestDB(core.Node{}))
+	Describe("executing watcher", func() {
+		var (
+			errs            chan error
+			mockFetcher     *mocks.MockStorageFetcher
+			mockQueue       *mocks.MockStorageQueue
+			mockTransformer *mocks.MockStorageTransformer
+			row             utils.StorageDiffRow
+			rows            chan utils.StorageDiffRow
+			storageWatcher  watcher.StorageWatcher
+		)
 
-		assert(func(err error) {
-			Expect(err).To(BeNil())
-			Expect(mockTailer.TailCalled).To(BeTrue())
-		}, w, mockTailer, []*tail.Line{})
-	})
+		BeforeEach(func() {
+			errs = make(chan error)
+			rows = make(chan utils.StorageDiffRow)
+			address := common.HexToAddress("0x0123456789abcdef")
+			mockFetcher = mocks.NewMockStorageFetcher()
+			mockQueue = &mocks.MockStorageQueue{}
+			mockTransformer = &mocks.MockStorageTransformer{Address: address}
+			row = utils.StorageDiffRow{
+				Id:           1337,
+				Contract:     address,
+				BlockHash:    common.HexToHash("0xfedcba9876543210"),
+				BlockHeight:  0,
+				StorageKey:   common.HexToHash("0xabcdef1234567890"),
+				StorageValue: common.HexToHash("0x9876543210abcdef"),
+			}
+		})
 
-	It("returns error if row parsing fails", func() {
-		mockTailer := fakes.NewMockTailer()
-		w := watcher.NewStorageWatcher(mockTailer, test_config.NewTestDB(core.Node{}))
-		line := &tail.Line{Text: "oops"}
+		It("logs error if fetching storage diffs fails", func(done Done) {
+			mockFetcher.ErrsToReturn = []error{fakes.FakeError}
+			storageWatcher = watcher.NewStorageWatcher(mockFetcher, test_config.NewTestDB(test_config.NewTestNode()))
+			storageWatcher.Queue = mockQueue
+			storageWatcher.AddTransformers([]transformer.StorageTransformerInitializer{mockTransformer.FakeTransformerInitializer})
+			tempFile, fileErr := ioutil.TempFile("", "log")
+			Expect(fileErr).NotTo(HaveOccurred())
+			defer os.Remove(tempFile.Name())
+			logrus.SetOutput(tempFile)
 
-		assert(func(err error) {
-			Expect(err).To(HaveOccurred())
-			Expect(err).To(MatchError(utils.ErrRowMalformed{Length: 1}))
-		}, w, mockTailer, []*tail.Line{line})
-	})
+			go storageWatcher.Execute(rows, errs, time.Hour)
 
-	It("logs error if no transformer can parse storage row", func() {
-		mockTailer := fakes.NewMockTailer()
-		address := common.HexToAddress("0x12345")
-		line := getFakeLine(address.Bytes())
-		w := watcher.NewStorageWatcher(mockTailer, test_config.NewTestDB(core.Node{}))
-		tempFile, err := ioutil.TempFile("", "log")
-		defer os.Remove(tempFile.Name())
-		Expect(err).NotTo(HaveOccurred())
-		logrus.SetOutput(tempFile)
-
-		assert(func(err error) {
-			Expect(err).NotTo(HaveOccurred())
-			logContent, readErr := ioutil.ReadFile(tempFile.Name())
-			Expect(readErr).NotTo(HaveOccurred())
-			Expect(string(logContent)).To(ContainSubstring(utils.ErrContractNotFound{Contract: address.Hex()}.Error()))
-		}, w, mockTailer, []*tail.Line{line})
-	})
+			Eventually(func() (string, error) {
```

Comment:
Good question!

As I understand it, [`Eventually`](https://onsi.github.io/gomega/#making-asynchronous-assertions) means the assertion only fails if it remains false through the timeout (but not just because it's false when it's first evaluated), and [`Done`](https://onsi.github.io/ginkgo/#asynchronous-tests) makes the tests run asynchronously so that channel reads aren't blocking other tests. 

I think it may be a little redundant to be using both `Eventually` and `Done`, since both enforce timeouts. We could probably get away with removing `Done`, but it seems like maybe there's added value insofar as the tests don't block while awaiting something like a (nanosecond :joy:) ticker.